File: examples/python/documentation/providers.py
# Copyright (c) HashiCorp, Inc.
# SPDX-License-Identifier: MPL-2.0

# DOCS_BLOCK_START:providers-import-providers
from constructs import Construct
from cdktf import App, TerraformStack
from imports.aws.provider import AwsProvider
from imports.aws.instance import Instance

class SimpleProviderStack(TerraformStack):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        AwsProvider(self, "aws",
            region = "us-east-1",
        )

        Instance(self, "hello",
            ami = "ami-2757f631",
            instance_type = "t2.micro",
        )

# DOCS_BLOCK_END:providers-import-providers
'''
DOCS_BLOCK_START:providers-import-providers
app = App()
SimpleProviderStack(app, "provider-stack")
app.synth()
DOCS_BLOCK_END:providers-import-providers
'''

# DOCS_BLOCK_START:providers-import-classes
import os
from constructs import Construct
from cdktf import App, TerraformStack, TerraformVariable, Token
from imports.aws.provider import AwsProvider
from imports.dnsimple.provider import DnsimpleProvider
from imports.dnsimple.zone_record import ZoneRecord

class ProviderStack(TerraformStack):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        AwsProvider(self, "aws",
            region = "us-east-1",
        )

        instance = Instance(self, "Hello",
            ami = "ami-2757f631",
            instance_type = "t2.micro",
        )

        dnsimpleToken = TerraformVariable(self, "dnsimpleToken", 
            type = "string",
            description = "dnsimple token",
            sensitive = True
        )

        dnsimpleAccount = TerraformVariable(self, "dnsimpleAccount",
            type = "string",
            description = "dnsimple account",
            sensitive = True
        )

        DnsimpleProvider(self, "dnsimple",
            token = dnsimpleToken.string_value,
            account = dnsimpleAccount.string_value
        )

        ZoneRecord(self, "web-www",
            zone_name = "example.com",
            name = "web",
            value = instance.public_ip,
            type = "A"
        )

# DOCS_BLOCK_END:providers-import-classes
'''
DOCS_BLOCK_START:providers-import-classes
app = App()
ProviderStack(app, "provider-stack")
app.synth()
DOCS_BLOCK_END:providers-import-classes
'''

File: examples/typescript/documentation/providers.ts
// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
// DOCS_BLOCK_START:providers-import,providers-import-classes
import { Construct } from "constructs";
import { TerraformStack, TerraformVariable, Token } from "cdktf";
import { AwsProvider } from "@cdktf/provider-aws/lib/aws-provider";
import { Instance } from "@cdktf/provider-aws/lib/instance";
// DOCS_BLOCK_END:providers-import,providers-import-classes

// DOCS_BLOCK_START:providers-import-classes
import { DnsimpleProvider } from "@cdktf/provider-dnsimple/lib/provider";
import { ZoneRecord } from "@cdktf/provider-dnsimple/lib/zone-record";
// DOCS_BLOCK_END:providers-import-classes

// DOCS_BLOCK_START:providers-import,providers-import-classes

export class ProvidersStack extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AwsProvider(this, "aws", {
      region: "us-east-1",
    });

    const instance = new Instance(this, "Hello", {
      ami: "ami-2757f631",
      instanceType: "t2.micro",
    });

    // DOCS_BLOCK_END:providers-import,providers-import-classes

    // DOCS_BLOCK_START:providers-import-classes
    const dnsimpleToken = new TerraformVariable(this, "dnsimpleToken", {
      type: "string",
      description: "dnsimple token",
      sensitive: true,
    });

    const dnsimpleAccount = new TerraformVariable(this, "dnsimpleAccount", {
      type: "string",
      description: "dnsimple account",
      sensitive: true,
    });

    new DnsimpleProvider(this, "dnsimple", {
      token: dnsimpleToken.stringValue,
      account: dnsimpleAccount.stringValue,
    });

    new ZoneRecord(this, "web-www", {
      zoneName: "example.com",
      name: "web",
      value: instance.publicIp,
      type: "A",
    });
    // DOCS_BLOCK_END:providers-import-classes

    // DOCS_BLOCK_START:providers-import,providers-import-classes
  }
}
// DOCS_BLOCK_END:providers-import,providers-import-classes
